packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3331,7 +3345,12 @@ export class ContainerRuntime
 	 */
 	public submitSignal(type: string, content: unknown, targetClientId?: string) {
 		this.verifyNotClosed();
-		const envelope = this.createNewSignalEnvelope(undefined /* address */, type, content);
+		const envelope = this.createNewSignalEnvelope(
+			undefined /* address */,
+			type,
```

Comment:
This is bugging me a little. Inside address is always undefined. I think this is the case for container internal messages only. So I expect targetClientId is never specified through here. If that is the case, then can we remove targetClientId parameter instead?

Reply:
If this is only showing up for testing, then this is not the path we want to support / test. Instead we'll want to use IFluidParentContext - go through ChannelCollection.

Reply:
> I think this is the case for container internal messages only

I do know that there is current external usage of container level signals (e.g. LiveShare uses container level signals for some of their signal use cases like LiveState and LivePresence: https://github.com/microsoft/live-share-sdk/blob/bd4eec01856542c13c944768320ec5988d089ba0/packages/live-share/src/interfaces.ts#L309). Maybe this is something we are moving away from, but I think they would immediately benefit. 

> So I expect targetClientId is never specified through here.

I was unaware of this, I thought initially DIS could've tentatively been using container runtime signals (which looking at the PR now doesn't seem like the case). Is there no value in sending targeted container level signals anymore? 

> If this is only showing up for testing, then this is not the path we want to support / test. Instead we'll want to use IFluidParentContext - go through ChannelCollection.

What type of tests should be added here? I see now that ChannelCollection/Datastore targeted signals address routing functionality are tested in localTestSignals. Both container and channelcollection share the `createNewSignal` function to create signal envelopes. I think the targeted signal **unit tests** are solely for telemetry testing, while the E2E tests actually validate that signals are begin targeted and routed properly.

Reply:
I believe this is location of Live Shares's acquisition: https://github.com/microsoft/live-share-sdk/blob/bd4eec01856542c13c944768320ec5988d089ba0/packages/live-share/src/schema-injection-utils.ts#L155C30-L155C46
That is a data object and not the container itself. So, don't need (want) to add this support here.
But I don't think the current tests you have are configured for use the real entrypoint we expect to be used. So, I would make a comment that `targetClientId` option there is only for test and support is to be removed. (Assuming I am right about not being able to adjust the tests.)

Reply:
I'm referencing the the LiveObjects that use the LiveObjectSynchronizer which are coordinated using container runtime signals: https://github.com/microsoft/live-share-sdk/blob/bd4eec01856542c13c944768320ec5988d089ba0/packages/live-share/src/internals/ContainerSynchronizer.ts#L221. I believe they have their own usage of 'DataObject' that is unrelated to ours.

I believe that Ryan might be the one to ask if they require this or not, but I can add the comment in the meantime.